controller/WFCORE-4770-boot-cli-script.adoc

```diff
@@ -0,0 +1,283 @@
+* `org.wildfly.internal.cli.boot.hook.script.output.file=<path to file to redirect output to>` (optional)
+* `org.wildfly.internal.cli.boot.hook.script.logging=<true to enable CommandContext CLI logging, false by default>` (optional)
+* `org.wildfly.internal.cli.boot.hook.script.warn.file=<path to file that contains warning message to display>` (optional)
+* `org.wildfly.internal.cli.boot.hook.script.error.file=<path to file that contains error message to display`> (optional)
+
+The service executes each line of the file. If an error occurs (Exception, error in error file, un-expected quit),
+the implementation Logger is used to display each line of the executed script that failed. In addition, the content of
+a non empty error, warn and output files are printed using the implementation Logger. An invalid condition in the service
+leads to a RuntimeException thrown, which will cause the server to abort booting.
+
+The CLI service disables the following commands commands:
+
+* `connect`
+* `reload`
+* `shutdown`
+* `embedded-*`
+* `clear`
+* additional commands found from to ServiceLoader.
+
+Calling any of the above commands will cause the the `AdditionalBootCliScriptInvoker` to fail and the server to abort.
+The raw operations `:reload` and `:shutdown` can still be called but is at the user's own risk.
+
+The CLI classes in use must not have any side effect on the server (eg: static initializer, static calls).
+
+==== Impact on log traces
+
+* The traces for CLI execution are mixed with the server traces (used to be printed before server starts). This is not
+a major issue, the traces are not interleaved. We have blocks of traces.
+* `WFLYSRV0025` displayed multiple times (up to 3 times). This will have an impact on tests that are only checking for
+this in the server log.
+* This is internal functionality to speed up the image start on OpenShift so it is not supported on bare metal (see
+Non-Requirements). If the system properties to trigger this are used, we will always log a WARN message saying this
+is only supported in the OpenShift images, and that any other usage is liable to change in the future.
+
+
+==== Impact in Openshift Builder image
+From a cloud user perspective, this can be summarized in few sentences:
+
+> If the EAP container launch script detects that some configuration is required, it starts the server in admin-mode.
+Once configuration is over, the server is reloaded  for the rare cases) into normal mode. All
+configurations aspects are handled internally by the EAP container. The extensions hook allows users to add their own
+custom CLI operations to the global server configuration.”
+
+It means that we now have a `CONFIG` phase (where the server runs in admin-only mode during the
+boot/configuration), followed by a `RUNNING` phase.
+
+New env vars:
+* `EXECUTE_BOOT_SCRIPT_INVOKER` - to control if the CLI boot time hook is to be used. Set to `true` in the builder image.
```

Comment:
Bullet lists is not displayed properly (missing a new line before the list)

Reply:
Fixed